### PR TITLE
Improve session/table property validation

### LINKED ISSUE #3793: CREATE TABLE does not validate table properties
This should fail, but it executes successfully:

```CREATE TABLE foo (x varchar) WITH (invalidproperty = 5)```

### PR DESCRIPTION
**Pull request by raghavsethi**
Fixes #3793 

Presto now validates and produces (more) useful error messages for:

- Invalid property names:
```
presto> CREATE TABLE test_raghavsethi_presto_5 (x varchar) WITH (foo = bar);
Query 20151023_211129_00000_kidpb failed: Table property 'foo' unsupported for catalog 'catalog'
```

- Incorrect types:
```
presto> CREATE TABLE test_raghavsethi_presto_5 (x varchar) WITH (format = true);
Query 20151023_223656_00001_nur8f failed: Unsupported value for table property 'format': Cannot convert 'true' to varchar
```

- Incorrect values:
```
presto> CREATE TABLE test_raghavsethi_presto_5 (x varchar) WITH (format = 'FOOBAR');
Query 20151023_223623_00000_nur8f failed: Unable to set table property 'format' to ''FOOBAR'': No enum constant com.facebook.presto.hive.HiveStorageFormat.FOOBAR
```

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/execution/SetSessionTask.java [hunk 1]
```diff
@@ -54,12 +57,18 @@ else if (propertyName.getParts().size() == 2) {
...
 
-        // verify the SQL value can be decoded by the property
-        metadata.getSessionPropertyManager().decodeProperty(propertyName.toString(), value, propertyMetadata.getJavaType());
+            // verify the SQL value can be decoded by the property
+            metadata.getSessionPropertyManager().decodeProperty(propertyName.toString(), value, propertyMetadata.getJavaType());
 
-        stateMachine.addSetSessionProperties(propertyName.toString(), value);
+            stateMachine.addSetSessionProperties(propertyName.toString(), value);
+        }
+        catch (SemanticException e) {
```
Comment (martint) [inline presto-main/src/main/java/com/facebook/presto/execution/SetSessionTask.java]: This `catch` is too broad. Which of the statements above can throw it?

Also:
* `SessionPropertyManager.decodeProperty` already validates the name, so this seems like an unrelated change
* The commit is supposed to be about validating table properties. This method handles setting session properties.

Comment (raghavsethi) [inline presto-main/src/main/java/com/facebook/presto/execution/SetSessionTask.java]: As part of fixing handling of table properties, we had to make the code that session properties used more generic. I'll update the commit message to reflect this.
Also, the error message does include details of the caught exception, so it should be pretty clear what went wrong.

File: presto-main/src/main/java/com/facebook/presto/metadata/TablePropertyManager.java [hunk 2]
```diff
@@ -80,4 +110,19 @@ public void addTableProperties(String catalog, List<PropertyMetadata<?>> tablePr
...
+    {
+        Object value = evaluateConstantExpression(expression, expectedType, metadata, session, ImmutableSet.of());
+
+        // convert to object value type of SQL type
+        BlockBuilder blockBuilder = expectedType.createBlockBuilder(new BlockBuilderStatus(), 1);
+        BlockUtils.appendObject(expectedType, blockBuilder, value);
+        Object objectValue = expectedType.getObjectValue(session.toConnectorSession(), blockBuilder, 0);
+
+        if (objectValue == null) {
+            throw new PrestoException(INVALID_TABLE_PROPERTY, "Session property value must not be null");
```
Comment (haozhun) [inline presto-main/src/main/java/com/facebook/presto/metadata/TablePropertyManager.java]: Wrong text

**Comment by haozhun** [comment]
Looks good. @erichwang please take a quick look.

**Comment by erichwang** [comment]
otherwise looks good
